snapcraft/internal/states/_pull_state.py

```diff
@@ -44,14 +44,16 @@ class PullState(State):
     yaml_tag = u'!PullState'
 
     def __init__(self, property_names, part_properties=None, project=None,
-                 stage_packages=None, build_packages=None):
+                 stage_packages=None, build_packages=None,
+                 source_details=None):
         # Save this off before calling super() since we'll need it
         # FIXME: for 3.x the name `schema_properties` is leaking
         #        implementation details from a higher layer.
         self.schema_properties = property_names
         self.assets = {
             'stage-packages': stage_packages,
             'build-packages': build_packages,
+            'source-details': source_details,
```

Comment:
I'm going to leave this as is for now.  The upcoming PR to add the source details to the snap's snapcraft.yaml takes advantage of dict.update().  We can discuss later.